src/main/resources/typescript-axios-webpack-module-federation/api.mustache

```diff
@@ -0,0 +1,13 @@
+/* tslint:disable */
+/* eslint-disable */
+
+import { APIFactory } from '@redhat-cloud-services/javascript-clients-shared/dist/utils';
+import type { AxiosInstance } from 'axios';
+import {
+{{>endpointList}}  } from './index';
+
+export const {{{clientName}}} = (BASE_PATH: string, instance: AxiosInstance) => {
+  return APIFactory(BASE_PATH, {
+{{>endpointList}}  }),
+  instance
```

Comment:
The instance is on the wrong line. It should be an argument of the API factory. The way it is written now it makes the instance part of a return statement. That's why the types are all messed up.